xls2xml/amp-t2d/T2D_xlsx.conf

```diff
@@ -0,0 +1,205 @@
+#AMP-T2D submission pack V2 spreadsheet.
+#
+#At the top level, "worksheets" defines a list of worksheets (titles) to be parsed.
+#Then for each worksheet, you define which fields to be parsed from this worksheet.
+#These fields are either "required" or "optional". "required" fields must be found
+#in the worksheet so that this worksheet is valid. Otherwise this worksheet is not
+#valid. The field names must match those values in the header row in the worksheet.
+#
+#This is a YAML file
+
+# List of worksheets to be parsed.
+worksheets:
+  - Project
+  - Cohort
+  - Phenotype_Trait
+  - Sample
+  - Attribute Mapping
+  - Analysis
+  - File
+
+# Worksheet specific requirements
+Project:
+  required:
+    - PI(s)
+    - PI email(s)
+    - Submitter Name(s)
+    - Submitter Email(s)
+    - Project Name
+    - Project Accronym
+    - Project Description
+    - Project Publications
+    - Total number of files being submitted
+
+  optional:
+    - Analysis covariates
+    - Funding acknowledgements
+    - Data URL
+    - Hold date
+    - Project website link
+    - Special analyses request
+    - External Links
+
+  data_type:
+    Total number of files being submitted: int
+
+Cohort:
+  required:
+    - Cohort ID
+    - Cohort Name
+    - Cohort Description
+    - Cohort Links
+    - Cohort Publications
+    - Case Control selection criteria
+    - External Links
+    - Ethnicity
+
+  optional:
+    - Link to cohort Logo
+    - "# T2D Cases"
+    - "# T2D Controls"
+    - Overlapping Projects
+    - Sample Collection Sites
+
+  data_type:
+    "# T2D Cases": int
+    "# T2D Controls": int
+
+Phenotype_Trait:
+  required:
+    - Sample_ID
+#    - ID
+#    - Age
+
+  optional:
+    - T2D_SELFREPORT
+    - T2D_HEALTH_PROVIDER
+    - T2D_MED
+    - T2D_FG
+    - T2D_HBA1C
+    - T2D_OGTT
+    - T2D_OTHER
+    - T2D_UNKNOWN
+    - T2D_MED_INSULIN
+    - T2D_MED_NONINSULIN
+    - AGE_T2D_DIAGNOSIS
+    - YRS_DIAGNOSIS_INS
+#    - NONT2D_FLAG
+    # Blood Biomarkers
+    - GLU_FAST
+    - INS_FAST
+    - GLU_2HR
+    - INS_2HR
+    - GLU_RANDOM
+    - CPEP_FAST
+    - HBA1C_PCT
+    - HBA1C_MMOL
+    - GAD_AB
+    - ISLET_AB
+    - ANTI_INSULIN_AB
+    - ZNT8_AB
+    - SERUM_CREATININE
+    - ADIPONECTIN
+    - LEPTIN
+    - CHOL
+    - LDL_DIRECT
+    - LDL_CALCULATED
+    - HDL
+    - TG
+    - LIPID_MED_LIPIDS
+    - STATIN_LIPIDS
+    # Anthropometry
+    - HEIGHT_CM
+    - WEIGHT_KG
+    - HIPC_CM
+    - WAISTC_CM
+    # Blood pressure and hypertension
+    - SBP
+    - DBP
+    - HYPERTENSION
+    - HTN_MED_BP
+    # Urine Measures
+    - CREAT_URINARY
+    - ALB_URINARY
+    - ACR_URINARY
+    # Smoking Status
+    - CURRENT_SMOKE
+    # Reproductive and Exogenous Hormone Use
+    - MENOPAUSE_STATUS
+    - HORMONES_ANY
+    - HORMONES_MENO
+
+#  data_type:
+#    Age: float
+
+Sample:
+  required:
+    - Sample_ID
+    - Subject_ID
+    - Geno_ID
+    - Phenotype
+    - Gender
+    - Analysis_alias
+    - Cohort ID
+    - Ethnicity
+
```

Comment:
Geno_ID is optional unless the sample_id is different in the VCF file - so in this case I assume it would be labelled as optional?

Reply:
The "Geno_ID" here is the header. In the "Geno_ID" column there could be empty values. It is "required" here just to check this column header is present in the worksheet.